src/main/java/com/direwolf20/buildinggadgets/common/tools/NetworkExtractor.java

```diff
@@ -0,0 +1,70 @@
+package com.direwolf20.buildinggadgets.common.tools;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.raoulvdberge.refinedstorage.api.network.INetwork;
+import com.raoulvdberge.refinedstorage.api.util.Action;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+public abstract class NetworkExtractor implements IItemHandler {
+    private final List<ItemStack> stacks;
+
+    protected NetworkExtractor(Collection<ItemStack> stacks) {
+        this.stacks = new ArrayList<>(stacks);
```

Comment:
```suggestion
        this.stacks = stacks.stream().filter(a->a!=null).collect(ImmutableList.toImmutableList());
```
We could then remove the need for null checks and in my opinion ImmutableLists are always nice... (and efficient)

Reply:
Ok, I made it an immutable list. I removed the null check in getStackInSlot because it was never necessary, and I also removed the null filler from the stream collection for the same reason. The stacks RS has in the cached list are non-null, as ItemStacks can generally be assumed to be.

The only reason I needed (and still need) a null check in extractItem is because INetwork#extractItem can return a null stack.